jpa/subsystem/src/main/java/org/jboss/as/jpa/config/Configuration.java

```diff
@@ -281,8 +282,15 @@ public static boolean needClassFileTransformer(PersistenceUnitMetadata pu) {
         if (pu.getProperties().containsKey(Configuration.JPA_CONTAINER_CLASS_TRANSFORMER)) {
             return Boolean.parseBoolean(pu.getProperties().getProperty(Configuration.JPA_CONTAINER_CLASS_TRANSFORMER));
         }
+        final String provider = pu.getPersistenceProviderClassName();
+        if (provider != null && provider.contains(HIBERNATE)) {
+            return false;
+        }
         return true;
     }
+    private static boolean isHibernateProvider(String provider) {
+        return provider == null || provider.contains("Hibernate");
```

Comment:
Why not use the constant here? Also shouldn't it be:
```java
return provider != null && provider.contains(HIBERNATE);
```

That would match what's above. If not, this can be deleted as I don't see it used anywhere.

Reply:
thanks for catching ^

Reply:
I'll explain why in a comment.